src/MasterSite.php

```diff
@@ -195,7 +195,7 @@ function ( $block_content, $block ) {
 		add_action( 'init', [ $this, 'login_redirect' ], 1 );
 		add_filter( 'attachment_fields_to_edit', [ $this, 'add_image_attachment_fields_to_edit' ], 10, 2 );
 		add_filter( 'attachment_fields_to_save', [ $this, 'add_image_attachment_fields_to_save' ], 10, 2 );
-		add_action( 'init', [ $this, 'p4_register_core_image_block' ] );
+		add_filter( 'register_block_type_args', [ $this, 'register_core_blocks_callback' ] );
```

Comment:
How can this filter work for WP < 5.5 if it was [only added in 5.5](https://developer.wordpress.org/reference/hooks/register_block_type_args/)? :thinking:

Reply:
Oh, because I'm extremely dumb and tested on the wrong version :facepalm:

Reply:
We can merge it to the `wp66` branch if that helps

Reply:
There's a fix for this on `wp55` https://github.com/greenpeace/planet4-master-theme/blob/wp55/src/MasterSite.php#L1099 that is not on `wp56` https://github.com/greenpeace/planet4-master-theme/blob/wp56/src/MasterSite.php#L1099, we might be missing some stuff or rebase ? @Inwerpsel do you have preferences on the fix ? I feel unregister/register is more cumbersome, but it might also be more efficient because it's probably executed less times in a process than `register_block_type_args`

Reply:
@lithrel your fix makes a lot more sense than the one on `wp55`, I wasn't aware of that new filter when I made that.

I think the easiest for us would be to keep the current `register_block_type` filter, but wrap it in a WP version check (WP < 5.5). We can already add the new filter, since it won't be called in WP < 5.5.

That way we can merge to master, don't have this change pending, and can switch WP versions locally without needing to check out a different branch of one of the repos.

As for efficiency, it gets called just once per request, and doesn't do anything expensive. If you want to be sure you can measure it, but I'm pretty sure it's negligible.

Reply:
Just to avoid confusion, I rebased [wp56](https://github.com/greenpeace/planet4-master-theme/tree/wp56) with all the relevant commits, and deleted `wp55`.